shared/js/moz_intl.js

```diff
@@ -187,36 +199,257 @@ global.mozIntl = {
+   * * maxUnit - (default) year
+   * * unit - second | minute | hour | day | week | month | year
+   *          (default) bestFit
+   *
+   * Example:
+   *
+   * const formatter = new Intl.RelativeTimeFormat(navigator.languages, {
+   *   unit: 'bestFit',
+   *   style: long'
+   * });
+   *
+   * var ms = Date.now() - 2 * 1000;
+   *
+   * formatter.format(ms); // "2 seconds ago"
+   *
+   * @param {Array} An array of languages
+   * @param {Array} Options object
+   * @returns An object with a formatter that returns Promises of strings
+   */
+  RelativeTimeFormat: function(locales, options) {
+    return {
+      /*
+       * ECMA 402 rev 3., 1.3.4, FormatRelativeTime
+       *
+       * Notes: This is a modified version of the function to use mozL10n
+       * and simplified to match current data set in data.properties
+       */
+      format: function(x) {
+        const ms = x - Date.now();
+        const units = computeTimeUnits(ms);
+
+        const unit = options.unit === 'bestFit' ?
+          getBestMatchUnit(units) : options.unit;
+
+        const v = units[unit];
+
+        // CLDR uses past || future
+        const tl = v < 0 ? '-ago' : '-until';
+        const style = options.style || 'long';
+
+        const entry = unit + 's' + tl + '-' + style;
+
+        return navigator.mozL10n.formatValue(entry, {
+          value: Math.abs(v)
+        });
+      },
+    };
+  },
+
+  _gaia: {
+    // This is an internal Firefox OS function, not part of the future standard
+    relativePart: function(milliseconds) {
+      const units = computeTimeUnits(milliseconds);
+      const unit = getBestMatchUnit(units);
+      return {
+        unit: unit + 's',
+        value: Math.abs(units[unit])
+      };
+    },
+
+    // This is an internal Firefox OS function, not part of the future standard
+    relativeDate: function(time, useCompactFormat, maxDiff) {
+      maxDiff = maxDiff || 86400 * 10; // default = 10 days
+      const secDiff = (Date.now() - time) / 1000;
+      if (isNaN(secDiff)) {
+        return navigator.mozL10n.formatValue('incorrectDate');
+      }
+
+      if (secDiff > maxDiff) {
+        const dateString = new Date(time).toLocaleString(navigator.languages, {
+          year: 'numeric',
+          month: 'numeric',
+          day: 'numeric'
+        });
+        return Promise.resolve(dateString);
+      }
+
+      const formatter = global.mozIntl.RelativeTimeFormat(navigator.languages, {
+        unit: 'bestFit',
+        style: useCompactFormat ? 'short' : 'long',
+        minUnit: 'minute',
+      });
+
+      return formatter.format(time);
+    },
   }
 };
 
+/*
+ * This data is used by DurationFormat
+ */
+const durationUnits = [
+  ['year', 3600000 * 24 * 365], // 1000 * 60 * 60 * 24 * 365
+  ['month', 3600000 * 24 * 30], // 1000 * 60 * 60 * 24 * 30
+  ['week', 3600000 * 24 * 7], // 1000 * 60 * 60 * 24 * 7
+  ['day', 3600000 * 24], // 1000 * 60 * 60 * 24
+  ['hour', 3600000], // 1000 * 60 * 60
```

Comment:
Or maybe not.  These expressions are probably pre-computed on code minification anyways.